components/fxa-client/src/internal/close_tabs.rs

```diff
@@ -92,3 +144,403 @@ impl FirefoxAccount {
         self.state.clear_commands_data(close_tabs::COMMAND_NAME);
     }
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
The tests are the biggest part of this patch, by quite a margin! 😅

Reply:
definitely love to see it though!!